src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,222 @@
+package com.google.sps.servlets.second;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public final class DataServletTest {
+
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private addTermRating newTermRating;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    newTermRating = new addTermRating();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+
+  @Test
+  public void AddingNewTermRating_NoTranslation() {
+
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+
+    // First Term Rating for this term.
+    when(request.getParameter("term-input")).thenReturn("I do not like this.");
+    when(request.getParameter("rating-term")).thenReturn("1");
+    when(request.getParameter("hoursOfWork")).thenReturn("8");
+    when(request.getParameter("difficulty")).thenReturn("4");
+    when(request.getParameter("prof-input")).thenReturn("The professor was amazing.");
+    when(request.getParameter("rating-professor")).thenReturn("3");
+    when(request.getParameter("languages")).thenReturn("false");
+    when(request.getParameter("ID")).thenReturn("numberOneId");
+    Entity termEntity = new Entity("Term");
+    Key termKey = termEntity.getKey();
+    when(request.getParameter("Course"))
+        .thenReturn(Course("Spring 2020", "Smith", 12, termKey));
+
+    newTermRating.addTermRating(request);
+
+    Entity termRatingQuery =
+        queryEntities("Rating", "reviewer-id", request.getParameter("reviewer-id")).get(0);
+    Entity expectedTermRatingEntity = new Entity("Rating");
+    expectedTermRatingEntity.setProperty("comments-term", "I do not like this.");
```

Comment:
This would be easier to follow if you set it right next to your mock response. Then I would know at a glance that they're the same

Reply:
It's also a bit weird creating an expectedTermRatingEntity when you go on to just check each value individually. Probably simpler to just check the value directly if you can't compare actual entities?

Reply:
Speaking of which, I'm guessing you've checked already, but you can't just compare expectedTermRatingEntity and termRatingQuery?